### ISSUE TITLE
Installation issue: Julia needs SSL certificates

### ISSUE THREAD

**Issue by JBlaschke** [LLNL/spack#32085]
### Steps to reproduce the issue

Some Julia packages need SSL certificates to be installed to: ` ${JULIA_DIR}/share/julia/cert.pem` -- which does not exist when Julia is installed using Spack -- eg:
```console
$ spack install julia
<wait for install to complete>
$ julia
] add IJulia
```

In this case the spec I'm trying to build is:
```console
julia%gcc@12.1.0 ^llvm%gcc@9.4.0
```
But I don't think it matters, as long as we're trying to install a reasonably modern Julia

### Error message

```julia
Failed to precompile IJulia [7073ff75-c697-5162-941a-fcdaad2a7d2a] to /home/blaschke/.julia/compiled/v1.7/IJulia/jl_3mPQBP.
ERROR: LoadError: InitError: SystemError: opening file "/home/blaschke/local/resources/spack/opt/spack/linux-ubuntu20.04-icelake/gcc-12.1.0/julia-1.7.3-nm4jvxihvfsinwikry2mmaag5mszoxbl/share/julia/cert.pem": No such file or directory
Stacktrace:
<...>
```

### Information on your system

```
* **Spack:** 0.19.0.dev0 (1cb4b2337f9246e13c1c5ccaa05b8576e7e2445a)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-icelake
* **Concretizer:** clingo
```

### Additional information

The solution is described here: https://discourse.julialang.org/t/package-fetch-issues-with-internal-ssl-certificate/43026
TL;DR on my wee ubuntu laptop I can just manually make the symlink:
```console
$ cd $(spack location -i julia)/share/julia
$ ln -s /etc/ssl/certs/ca-certificates.crt cert.pem
```
This is only a stopgap solution -- ideally we either should fetch `cert.pem` from an official source, or perform the link step above after the installation step. I researched this a wee bit, and found: `run_after("install")`.

@glennpj @haampie @vchuravy what do you think? I am happy to put together a PR, but I would like to hear your opinion as to which would be the best solution. Also: I know my solution works for linux (not even sure if it works for all linux'es), is it ok to include a partial solution now and add more OS's as we go?

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by RemiLacroix-IDRIS** [comment]
Hi. I'm running into the same issue. Any news?

I guess fetching `cert.pem` from an official source would be the best solution.

Edit: In fact the OpenSSL package seems to already be doing something clever so linking to its `cert.pem` is actually probably the best solution.
